DIPs/DIP1028.md

```diff
@@ -50,11 +54,42 @@ compiler switch:
 
 There are no grammar changes.
 
+### Non-Virtual Templated Functions
+
+Non-virtual Function templates get their attributes inferred if they are not explicitly marked.
+Hence, this proposal should not affect them.
+
 ## Breaking Changes and Deprecations
 
 This will likely break most code that has not already been annotated with `@safe`,
-`@trusted`, or `@system`. Fortunately, the solution is easy, although tedious: annotate
-functions that aren't safe with `@trusted` or `@system`.
+`@trusted`, or `@system`. Annotate functions that aren't safe with `@system`.
+Once the project compiles again successfully, start with the leaves market `@system` and
+modify as required to make them `@safe`.
+
+In general, `@system` should not be applied to blocks of functions. It should be added
+specifically to each function that requires it.
+
+### Extern C and C++ Functions
+
+An unmarked `extern (D)` function will now be considered `@safe`. If its implementation is not recompiled,
+it will no longer link because the mangling will be different. But for `extern (C)` and `(C++)` functions,
+safety is not part of the mangling and it will compile and link without error. This has always
+relied on the user getting it correct.
+
+Interfaces to extern C and C++ functions should always be explicitly marked.
+
+### 3rd Party Libraries
+
+It's best practice for 3rd party libraries to use explicit safety annotations and not rely on defaults.
+It is also best practice to compile those libraries with the same compiler that is being used
+to compile the project.
```

Comment:
This isn't practical for closed source libraries. It also creates a huge burden on individuals.

Compare that with Rust:

https://doc.rust-lang.org/edition-guide/editions/index.html

These breaking changes should be put together into a different version, and you should be able to choose in dub and on the command line what version to use. Rather than simply telling people to update all their code and to just use the same compiler without any sort of proper version control on D's end like many many other languages have.